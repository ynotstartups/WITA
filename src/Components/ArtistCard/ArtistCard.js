import React, { useState } from "react";
import { withStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import Card from "@material-ui/core/Card";
import CardMedia from "@material-ui/core/CardMedia";
import CardHeader from "@material-ui/core/CardHeader";
import CardActions from "@material-ui/core/CardActions";
import CardActionArea from "@material-ui/core/CardActionArea";
import IconButton from "@material-ui/core/IconButton";
import { connect } from "react-redux";
import { saveArtist, removeArtist } from "../../redux/actions";
import Favorite from "@material-ui/icons/Favorite";
import gql from "graphql-tag";
import { withRouter } from "react-router-dom";

function mapStateToProps({ savedArtists }) {
  return {
    savedArtists: savedArtists
  };
}

function mapDispatchToProps(dispatch) {
  return {
    // TODO can I just pass id into redux actions?
    saveArtist: id => dispatch(saveArtist(id)),
    removeArtist: id => dispatch(removeArtist(id))
  };
}

const ArtistCard = ({
  imageUrl,
  displayLabel,
  id,
  href,
  saveArtist,
  removeArtist,
  savedArtists,
  classes,
  history
}) => {
  const [isSaved, setIsSaved] = useState(savedArtists.includes(id));

  const handleClick = () => {
    if (isSaved) {
      removeArtist(id);
    } else {
      saveArtist(id);
    }
    setIsSaved(!isSaved);
  };
  return (
    <Card className={classes.card}>
      <CardHeader
        action={
          <IconButton onClick={handleClick}>
            <Favorite color={isSaved ? "secondary" : "inherit"} />
          </IconButton>
        }
        title={displayLabel}
      />
      <CardActionArea
        onClick={() => {
          history.push(`/gallery/${id}`);
        }}
      >
        <CardMedia className={classes.image} image={imageUrl} />
      </CardActionArea>
      <CardActions>
        <Button
          size="small"
          color="primary"
          onClick={() => {
            window.open(`https://www.artsy.net${href}`);
          }}
        >
          Artsy
        </Button>
        <Button
          size="small"
          color="primary"
          onClick={() => {
            window.open(
              `https://en.wikipedia.org/wiki/${displayLabel.replace(/ /g, "_")}`
            );
          }}
        >
          Wikipedia
        </Button>
        <Button
          size="small"
          color="primary"
          onClick={() => {
            window.open(
              `https://www.google.com/search?tbm=isch&q=${displayLabel}`
            );
          }}
        >
          Google
        </Button>
      </CardActions>
    </Card>
  );
};

ArtistCard.fragments = {
  ArtistCardArtist: gql`
    fragment ArtistCardArtist on Artist {
      id
      name
      href
      artworks(page: 1) {
        displayLabel
        images {
          url
          title
        }
      }
    }
  `
};

const styles = theme => ({
  card: {
    display: "flex",
    height: "100%",
    flexDirection: "column",
    justifyContent: "space-between"
  },
  image: {
    // LOL Artsy cropped it to 230
    height: 230,
    width: 230,
    margin: "auto",
    backgroundSize: "cover",
    backgroundPosition: "center",
    backgroundRepeat: "no-repeat"
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withRouter(withStyles(styles)(ArtistCard)));
