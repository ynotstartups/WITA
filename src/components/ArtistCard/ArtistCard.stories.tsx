import React from "react"
import { storiesOf } from "@storybook/react"

storiesOf(`Dashboard/Header`, module).add(`default`, () => (
  <a href="whatever.com" style={{background: "black", height: 200, width: 200, display: "flex", alignItems: "center", justifyContent: "center"}}>
    <div style={{ color: "white", border: "solid 3px white", padding: 10}}>Impressionism</div>
  </a>
))
