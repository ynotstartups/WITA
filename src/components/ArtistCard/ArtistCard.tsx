import React, { Children } from "react"
import { withStyles, createStyles, WithStyles } from "@material-ui/core/styles"
import Button from "@material-ui/core/Button"
import Card from "@material-ui/core/Card"
import CardMedia from "@material-ui/core/CardMedia"
import CardHeader from "@material-ui/core/CardHeader"
import CardActions from "@material-ui/core/CardActions"
import CardActionArea from "@material-ui/core/CardActionArea"
import { navigate } from "@reach/router"

import ArtistSaveButton from "./ArtistSaveButton/ArtistSaveButton"

interface AnchorButtonProps {
  href: string
  children: string
}

const AnchorButton: React.FunctionComponent<AnchorButtonProps> = ({
  href,
  children,
}) => {
  return (
    <Button
      size="small"
      color="primary"
      rel="noopener noreferrer"
      target="_blank"
      href={href}
      style={{ color: "black" }}
    >
      {children}
    </Button>
  )
}

const styles = () =>
  createStyles({
    card: {
      display: "flex",
      height: "100%",
      flexDirection: "column",
      justifyContent: "space-between",
      border: "black 1px solid",
    },
    image: {
      // LOL Artsy cropped it to 230
      height: 230,
      width: 230,
      margin: "auto",
      backgroundSize: "cover",
      backgroundPosition: "center",
      backgroundRepeat: "no-repeat",
    },
  })

interface ArtistCardProps extends WithStyles<typeof styles> {
  imageUrl?: string
  displayLabel: string
  id: string
  href: string
}

const ArtistCard: React.FunctionComponent<ArtistCardProps> = ({
  imageUrl = "",
  displayLabel,
  id,
  href,
  classes,
}) => (
  <Card className={classes.card} elevation={2}>
    <CardHeader action={<ArtistSaveButton id={id} />} title={displayLabel} />
    <CardActionArea
      onClick={() => {
        navigate(`/gallery/${id}`)
      }}
    >
      <CardMedia
        className={classes.image}
        image={imageUrl || `${process.env.PUBLIC_URL}/img/no-image.jpg`}
      />
    </CardActionArea>
    <CardActions>
      <AnchorButton href={`https://www.artsy.net${href}`}>Artsy</AnchorButton>
      <AnchorButton
        href={`https://en.wikipedia.org/wiki/${displayLabel.replace(
          / /g,
          "_"
        )}`}
      >
        Wikipedia
      </AnchorButton>
      <AnchorButton
        href={`https://www.google.com/search?tbm=isch&q=${displayLabel}`}
      >
        Google
      </AnchorButton>
    </CardActions>
  </Card>
)

export default withStyles(styles)(ArtistCard)
