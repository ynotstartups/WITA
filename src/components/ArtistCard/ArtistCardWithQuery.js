import React from "react"
import gql from "graphql-tag"
import { Query } from "react-apollo"
import PropTypes from "prop-types"

import LoadingSpinner from "../LoadingSpinner/LoadingSpinner"
import ArtistCard from "./ArtistCard"

const GET_ARTIST = gql`
  query Artist($id: String!) {
    artist(id: $id) {
      id
      displayLabel
      href
      imageUrl
    }
  }
`

function ArtistCardWithQuery({ id }) {
  return (
    <Query query={GET_ARTIST} variables={{ id }}>
      {({ loading, error, data }) => {
        if (loading) return <LoadingSpinner />
        if (error) return `Error! ${error.message}`
        return <ArtistCard {...data.artist} />
      }}
    </Query>
  )
}

export default ArtistCardWithQuery

ArtistCardWithQuery.propTypes = {
  id: PropTypes.string.isRequired,
}
