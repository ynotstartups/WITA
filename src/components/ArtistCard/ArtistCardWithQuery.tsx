import React from "react"
import gql from "graphql-tag"
import { Query } from "react-apollo"

import LoadingSpinner from "../LoadingSpinner/LoadingSpinner"
import ArtistCard from "./ArtistCard"

const GET_ARTIST = gql`
  query Artist($id: String!) {
    artist(id: $id) {
      id
      displayLabel
      href
      imageUrl
    }
  }
`

interface Props {
  id: String
}

const ArtistCardWithQuery: React.FunctionComponent<Props> = ({ id }) => {
  return (
    <Query query={GET_ARTIST} variables={{ id }}>
      {({ loading, error, data }) => {
        if (loading) return <LoadingSpinner />
        if (error) return `Error! ${error.message}`
        return <ArtistCard {...data.artist} />
      }}
    </Query>
  )
}

export default ArtistCardWithQuery
