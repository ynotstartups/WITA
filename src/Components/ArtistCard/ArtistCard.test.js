import React from "react";
import renderer from "react-test-renderer";
import ArtistCard from "./ArtistCard";
import { MemoryRouter } from "react-router";

jest.mock("./ArtistSaveButton/ArtistSaveButton");

test("ArtistCard renders correctly", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <ArtistCard
          imageUrl={"artsy"}
          displayLabel={"Tiger"}
          id={"tiger"}
          href={"wow"}
        />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
