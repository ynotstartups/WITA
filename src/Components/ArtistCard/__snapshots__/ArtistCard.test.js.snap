// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistCard renders correctly 1`] = `
<div
  className="MuiPaper-root-4 MuiPaper-elevation1-7 MuiPaper-rounded-5 MuiCard-root-3 ArtistCard-card-1"
>
  <div
    className="MuiCardHeader-root-31"
  >
    <div
      className="MuiCardHeader-content-34"
    >
      <span
        className="MuiTypography-root-37 MuiTypography-headline-42 MuiCardHeader-title-35"
      >
        Tiger
      </span>
    </div>
    <div
      className="MuiCardHeader-action-33"
    >
      <div />
    </div>
  </div>
  <button
    className="MuiButtonBase-root-76 MuiCardActionArea-root-73"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <div
      className="MuiCardMedia-root-79 ArtistCard-image-2"
      style={
        Object {
          "backgroundImage": "url(\\"artsy\\")",
        }
      }
    />
    <span
      className="MuiCardActionArea-focusHighlight-75"
    />
    <span
      className="MuiTouchRipple-root-110"
    />
  </button>
  <div
    className="MuiCardActions-root-81"
  >
    <button
      className="MuiButtonBase-root-76 MuiButton-root-84 MuiButton-text-86 MuiButton-textPrimary-87 MuiButton-flat-89 MuiButton-flatPrimary-90 MuiButton-sizeSmall-107 MuiCardActions-action-83"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-85"
      >
        Artsy
      </span>
      <span
        className="MuiTouchRipple-root-110"
      />
    </button>
    <button
      className="MuiButtonBase-root-76 MuiButton-root-84 MuiButton-text-86 MuiButton-textPrimary-87 MuiButton-flat-89 MuiButton-flatPrimary-90 MuiButton-sizeSmall-107 MuiCardActions-action-83"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-85"
      >
        Wikipedia
      </span>
      <span
        className="MuiTouchRipple-root-110"
      />
    </button>
    <button
      className="MuiButtonBase-root-76 MuiButton-root-84 MuiButton-text-86 MuiButton-textPrimary-87 MuiButton-flat-89 MuiButton-flatPrimary-90 MuiButton-sizeSmall-107 MuiCardActions-action-83"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-85"
      >
        Google
      </span>
      <span
        className="MuiTouchRipple-root-110"
      />
    </button>
  </div>
</div>
`;
